compiler/rustc_hir/src/hir.rs

```diff
@@ -442,6 +442,7 @@ pub enum GenericBound<'hir> {
     Trait(PolyTraitRef<'hir>, TraitBoundModifier),
     // FIXME(davidtwco): Introduce `PolyTraitRef::LangItem`
     LangItemTrait(LangItem, Span, HirId, &'hir GenericArgs<'hir>),
+    Unsized(Span),
```

Comment:
I didn't have time to get to this. But I'm a bit concerned about this. Would like to have potentially seen this reuse `Trait` in some way

Reply:
I brought this up in https://github.com/rust-lang/rust/pull/85799#discussion_r675629404 but then forgot about it again entirely. Sorry, should've paid more attention. I'm fine doing a quick revert and figuring all this out in a new PR

Reply:
As you all prefer, but I am concerned about the size effects of adding a `Span` to `Trait`. @oli-obk, that comment was in a separate `enum`, to carry the span in the obligation, this is to carry it in the `HIR`.

Reply:
`PolyTraitRef` already contains a span, can we re-use that?

Reply:
Looking back at this, the reason I didn't want to just use the `Trait` variant were because we want this exclusively [for this](https://github.com/rust-lang/rust/pull/87421/files#diff-0a61b538a3cec072c76fecae4635af6a12ec3256860029ac70549c2aa53ab394R1448-R1456), as an "empty signal" and wanted to make sure we didn't try to evaluate these by accident (as they are already handled differently elsewhere). I _could_ have added a field to `Trait`, but I thought that would have caused the enum's size to increase (although that might not be the case, looking at the `LangItem` variant).